### PR TITLE
avm2: Fix the stub for `Vector` + some related cleanup

### PR DESCRIPTION
**Pull request by torokati44**
As per our (not so) recent Discord discussion.

EDIT: Plus some related cleanup following #7597.

### CODE REVIEW

**Comment by adrian17** [comment]
Can you also update `Vector` to `Vector.<Number>` in `ShaderEvent.as`? It should work after your first commit.

**Comment by torokati44** [comment]
Yep, done!

**Review by Herschel — APPROVED** [review]
Thank you!

File: core/src/avm2/globals/flash/events/ShaderEvent.as [hunk 1]
```diff
@@ -12,7 +12,7 @@ package flash.events
     {
         public static const COMPLETE:String = "complete"; // Defines the value of the type property of a complete event object.
 
-        public var vector: Vector; // The Vector.<Number> object that was passed to the ShaderJob.start() method.
+        public var vector: Vector.<Number>; // The Vector.<Number> object that was passed to the ShaderJob.start() method.
```
Comment (adrian17) [inline core/src/avm2/globals/flash/events/ShaderEvent.as:15]: same in the constructor :)

**Review by adrian17 — APPROVED** [review]

File: core/src/avm2/globals/flash/events/ShaderEvent.as (hunk 1 above)
Comment (torokati44) [reply core/src/avm2/globals/flash/events/ShaderEvent.as:15]: Bah, I made sure to do a search so I don't miss anything similar. And yet! In the very same file even! :facepalm: Thanks!

**Pull request merged**